save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProjectController.kt

```diff
@@ -334,6 +340,16 @@ class DownloadProjectController(private val configProperties: ConfigProperties,
             }
     }
 
+    // Notice: branches should contain explicit `origin/` prefix
+    private fun switchBranch(git: Git, repoUrl: String, branchOrCommit: String) {
+        log.debug("For $repoUrl switching to the $branchOrCommit")
+        try {
+            git.checkout().setName(branchOrCommit).call()
+        } catch (ex: RefNotFoundException) {
+            log.warn("Provided branch/commit $branchOrCommit wasn't found, will use default branch")
```

Comment:
I think we should crash execution completely in this case. Why spend additional resources on something, that user hasn't requested?

Reply:
If I remember correctly, we discussed with @akuleshov7, that in this case we will use default branch

Reply:
May be in some cases it will be unrequested, but since jgit require `/origin` prefix and since we are not able to check whether the branch name contain typos like, `maste` or `mmaster` etc, probably it will be also useful for such cases

Reply:
Let's leave it as a workaround until we can display preprocessing errors on FE (#331), but it shouldn't be default behavior IMO.